### PR TITLE
Updated the Java FAQ to include known limitations section

### PR DESCRIPTION
**Pull request by nrgraham23**
Added documentation for persistent problems involving the java
bindings and psm/psm2.  Problems involving psm/psm2 have arisen
on multiple occasions, this will hopefully advise users on
how to fix or get help with errors they encounter.

Signed-off-by: Nathaniel Graham <ngraham@lanl.gov>

@hppritcha can you take a look?

### CODE REVIEW

**Comment by hppritcha** [comment]
@jsquyres take a look.  i don't have write perms on this repo.

File: faq/java.inc [hunk 1]
```diff
@@ -128,3 +128,22 @@ C/Fortran code).
 We suggest users experiencing performance problems to contact the
 <a href=\"https://www.open-mpi.org/community/lists/ompi.php\">
 OMPI users mailing list</a>.";
+
+/////////////////////////////////////////////////////////////////////////
+
+$q[] = "What are the known limitations of the Java Bindings?";
+$anchor[] = "java_limitations";
+
+$a[] = "There are currently issues with Intel Truescale (PSM) and Omnipath
```
Comment (jsquyres) [inline faq/java.inc]: @nrgraham23 @hppritcha @matcabral Can you revise this a bit?  It's a bit weird to say "currently" in an FAQ item (because FAQ items live for a long, long time).  Can you be a bit more specific about which versions of PSM / PSM2 had problems, and which versions fixed the problem?

Also, there's inconsistencies about the capitalization of PSM in this entry -- can you make it consistent?

(I know this PR has already been merged, but it would be nice if a subsequent PR/commit could clarify / fix the language a bit -- thanks!)

Comment (nrgraham23) [inline faq/java.inc]: @jsquyres Do you have a suggestion for a better place to put the known issues section?  I can reword it to not use "currently", but this is intended to be a temporary warning for any users seeing this issue.
